torch/custom_class.h

```diff
@@ -17,52 +17,44 @@
+/// taking an `int` and a `std::string` as argument.
 template <class... Types>
 detail::types<void, Types...> init() {
   return detail::types<void, Types...>{};
 }
 
-// Entry point for custom C++ class registration. To register a C++ class
-// in PyTorch, instantiate `torch::class_` with the desired class as the
-// template parameter. Typically, this instantiation should be done in
-// the initialization of a global variable, so that the class will be
-// made available on dynamic library loading without any additional API
-// calls needed. For example, to register a class named Foo, you might
-// create a global variable like so:
-//
-//   static auto register_foo = torch::class_<Foo>("Foo")
-//     .def("myMethod", &Foo::myMethod)
-//     .def("lambdaMethod", [](const c10::intrusive_ptr<Foo>& self) {
-//       // Do something with `self`
-//     });
-//
-// In addition to registering the class, this registration also chains
-// `def()` calls to register methods. `myMethod()` is registered with
-// a pointer to the Foo class's `myMethod()` method. `lambdaMethod()`
-// is registered with a C++ lambda expression.
+/// Entry point for custom C++ class registration. To register a C++ class
+/// in PyTorch, instantiate `torch::class_` with the desired class as the
+/// template parameter. Typically, this instantiation should be done in
+/// the initialization of a global variable, so that the class will be
+/// made available on dynamic library loading without any additional API
+/// calls needed. For example, to register a class named Foo, you might
+/// create a global variable like so:
+///
+///     static auto register_foo = torch::class_<Foo>("Foo")
+///       .def("myMethod", &Foo::myMethod)
+///       .def("lambdaMethod", [](const c10::intrusive_ptr<Foo>& self) {
+///         // Do something with `self`
+///       });
+///
+/// In addition to registering the class, this registration also chains
+/// `def()` calls to register methods. `myMethod()` is registered with
+/// a pointer to the Foo class's `myMethod()` method. `lambdaMethod()`
+/// is registered with a C++ lambda expression.
 template <class CurClass>
 class class_ {
   static_assert(std::is_base_of<CustomClassHolder, CurClass>::value,
     "torch::class_<T> requires T to inherit from CustomClassHolder");
 
  public:
-  // Constructor. String argument className_ is the name you would like to
-  // see this class exposed as in Python and TorchScript. For example, if
-  // you pass in "MyStack" here, the class will appear as
-  // `torch.classes.MyStack` in both Python and TorchScript.
+  /// This constructor actually registers the class type.
+  /// String argument `className_` is the name you would like to
+  /// see this class exposed as in Python and TorchScript. For example, if
+  /// you pass in "MyStack" here, the class will appear as
+  /// `torch.classes.MyStack` in both Python and TorchScript.
   class_(std::string className_) : className(std::move(className_)) {
```

Comment:
stupid question - why is it className_ (with underscore)? :)

should the constructor be `explicit`?

Reply:
1) Dunno, horace wrote it like that. I can change it
2) yeah probably